packages/catalog-process/src/services/catalogService.ts

```diff
@@ -196,14 +184,36 @@ const hasNotDraftDescriptor = (eService: EService): void => {
   }
 };
 
-export const catalogService = {
-  async createEService(
+export class CatalogService {
+  private readModelService;
+  private repository;
+
+  constructor(
+    readModelService: ReadModelService,
+    config: CatalogProcessConfig
+  ) {
+    this.readModelService = readModelService;
+    this.repository = eventRepository(
```

Comment:
this is a dependency that should be injected like `ReadModelService`.
In general it is not a good practice to create connections inside a class constructor (while it is sometimes acceptable if the class purpose is to handle the connection itself).

With this change, you can apply [Zibo's](https://github.com/pagopa/interop-be-monorepo/pull/110#issuecomment-1830148738) and [Gabro's](https://github.com/pagopa/interop-be-monorepo/pull/110#issuecomment-1830190835) suggestions

Reply:
@galales `CatalogService` will be instantiated inside the router file. From a _responsibility separation_ point of view, I would keep the router _unaware_ of the `eventRepository` creation. What do you think?

Anyway, I am applying @MalpenZibo's and @gabro's suggestions and converting the classes into objects, but I am facing an issue because in some parts I need to pass a `ReadModelService` object as a parameter, and since it should no longer be a class, I don't know what type to use. Apparently the following line is not enough:
`export type ReadModelService = typeof readModelServiceBuilder`

Reply:
Try with `export type ReadModelService = ReturnType<typeof readModelServiceBuilder>`

Reply:
Thanks a lot @MalpenZibo, now it's working!

Reply:
that's a possible approach, but shouldn't be the same for the read model service?
So far we used dependency injection pattern, which resulted to be a bit more flexible for our use cases. I would keep using it for now.